src/main/java/io/jenkins/plugins/coverage/source/DefaultSourceFileResolver.java

```diff
@@ -80,11 +79,10 @@ public void resolveSourceFiles(Run<?, ?> run, FilePath workspace, TaskListener l
 
         listener.getLogger().printf("%d source files need to be copied%n", paints.size());
 
-        ExecutorService service = Executors.newFixedThreadPool(5);
         // wait until all tasks completed
         CountDownLatch latch = new CountDownLatch(paints.entrySet().size());
 
-        paints.forEach((sourceFilePath, paint) -> service.submit(() -> {
+        paints.forEach((sourceFilePath, paint) -> Timer.get().submit(() -> {
```

Comment:
Given this is inside a loop, I'm not sure this could affect performance and it would be better to fix the previous implementation by:
- Shutting down the executor.
- Adding a naming thread factory.

@jglick what's your opinion here?

Reply:
Without knowing a typical value for `paints.size()`, or how long each task runs, it is hard to say. (`Timer` is a global pool of 10 threads, intended for quick tasks.) Also, what is calling `resolveSourceFiles`—a build thread? From a brief glance at the code, I would think it would be better to just run these tasks serially, since they probably spend most time in `FilePath.copyTo(FilePath)` on the same agent, which cannot be usefully parallelized anyway. Would need some context from @cizezsy here.

Reply:
I just pushed a commit to implement Jesse's suggestion of doing it serially.

Reply:
I suppose you should edit the PR title accordingly.

Reply:
The `paints.size()` is the number of the source files that need to be copied. It will copy source files from the agent, render them and save the result to master. The time spent most in `FilePath.copyTo()` which will copy source files from the agent and `paintSourceCode()` which will render the source file and save it to master.

As `FilePath.copyTo()` cannot be parallelized, maybe we need to run `paintSourceCode()` parallelly to increase the performance? @jglick

Reply:
You could only see a speedup if the system were otherwise mostly idle, and you need to support build interruption. I doubt it is worth making the code more complicated for a speculative optimization, especially as build time is a low priority for performance compared to UI responsiveness.

Reply:
Yeah, you are right...approved this pr, thanks.